Solvers : Used checksysodesol in test_ode.py

sudz123 opened the pull request:
Partially addresses #15097 

#### Release notes

<!-- BEGIN RELEASE NOTES -->

NO ENTRY

<!-- END RELEASE NOTES -->

sudz123 committed:
Solvers : Used checksysodesolve in test_ode.py according to Issue #15097. Also slightly modified test solutions in the same function.

asmeurer commented:
The changes here look fine, but does this fully fix the issue? If not, the "fixes" should be removed from the PR description. CC @cbm755

cbm755 commented on sympy/solvers/tests/test_ode.py:
```diff
@@ -34,78 +34,47 @@ def test_linear_2eq_order1():
-    sol6 = [Eq(x(t), (C1*exp(Integral(2, t)) + C2*exp(-Integral(2, t)))*exp(Integral(5*t, t))), \
-    Eq(y(t), (C1*exp(Integral(2, t)) - C2*exp(-Integral(2, t)))*exp(Integral(5*t, t)))]
+    sol6 = [Eq(x(t), (C1*exp(2*t) + C2*exp(-2*t))*exp(2.5*t**2)), \
+    Eq(y(t), (C1*exp(2*t) - C2*exp(-2*t))*exp(2.5*t**2))]
```
`S(5)/2` or `Rational(5, 2)` please

cbm755 commented on sympy/solvers/tests/test_ode.py:
```diff
@@ -34,78 +34,47 @@ def test_linear_2eq_order1():
-    s = [(l.lhs, l.rhs) for l in s]
-    assert eq8[0].subs(s).doit().simplify()
-    assert eq8[1].subs(s).doit().simplify()
+    sol8 = [Eq(x(t), (C1*exp((sqrt(77)/2 + 9/2)*(t**3)/3) + \
```
`Rational(9, 2)`

cbm755 commented on sympy/solvers/tests/test_ode.py:
```diff
@@ -34,78 +34,47 @@ def test_linear_2eq_order1():
-    assert eq7[0].subs(s).doit().simplify()
-    assert eq7[1].subs(s).doit().simplify()
+    sol7 = [Eq(x(t), (C1*cos((t**3)/3) + C2*sin((t**3)/3))*exp(2.5*t**2)), \
+    Eq(y(t), (-C1*sin((t**3)/3) + C2*cos((t**3)/3))*exp(2.5*t**2))]
```
as above

cbm755 reviewed (commented):
@asmeurer  is right, it is only a partial fix for the issue.  I think you can just change the PR comment b/c the commit message doesn't have the magic "fixes ..." string.

Are there any other places in these tests that can get this fix?

sidhantnagpal commented on sympy/solvers/tests/test_ode.py:
```diff
@@ -34,78 +34,47 @@ def test_linear_2eq_order1():
+    assert checksysodesol(eq1, sol1) == (True, [0, 0])
 
     eq2 = (Eq(diff(x(t),t), 2*x(t) + 4*y(t)), Eq(diff(y(t),t), 12*x(t) + 41*y(t)))
     sol2 = [Eq(x(t), 4*C1*exp(t*(sqrt(1713)/2 + 43/2)) + 4*C2*exp(t*(-sqrt(1713)/2 + 43/2))), \
```
`S(43)/2` instead of `43/2` (Similar change as suggested [here](https://github.com/sympy/sympy/pull/15133#discussion_r212488513))

sudz123 committed:
Solvers : Modified test cases in test_ode.py

sudz123 commented:
I changed the expressions to S(x)/y. I'll look into more places where we can put this fix. For now, I have changed the comment "fixes" from the PR description.

sudz123 committed:
Solvers : Used checksysodesol at multiple locations in test_ode.py instead of current tests

cbm755 commented:
This looks good to me.

For future PRs:
   * use `solvers: used...` (no space before `:`)
   * some lines are too long.
Read about how to format commit msgs on the SymPy dev guide.  But no need to change these ones.

I now edit your release notes entry a bit...

cbm755 commented:
Actually, maybe this doesn't need release notes as its not user-facing in any way...

I will merge in 24 hours or so.

sudz123 commented:
Sure, will do. I am a bit new to Sympy, so still trying to get used to it. But thanks for the tips, will keep them in mind :)

cbm755 commented:
Are there any bigger systems (e.g., 3x3) this could use this change too?  You could do a difference PR as this one is approved and ready to merge.

sudz123 commented:
I'll look into that. But one thing I am a bit confused about is why the line `assert s == sol` gave false everytime in a isympy terminal but yet passes all the tests. Although, now with `checksysodesol` being used this shouldn't be an issue.

cbm755 commented:
> why the line assert s == sol gave false a lot of times in an isympy terminal 

File another bug for this?

sudz123 commented:
Sure I'll try to do that.
